test/expr/expressions.swift

```diff
@@ -829,5 +829,11 @@ func swift22_deprecation_increment_decrement() {
   _ = --si  // expected-warning {{'--' is deprecated: it will be removed in Swift 3}}
 }
 
-
-
+// SR-628 mixing lvalues and rvalues in tuple expression
+var x = 0
+var y = 1
+let _ = (x, x.successor()).0
+let _ = (x, 3).1
+(x,y) = (2,3)
+(x,4) = (1,2) // expected-error {{cannot assign to value: function call returns immutable value}}
```

Comment:
A better diagnostic would be ```{function call result|literal value|result of an expression} not allowed on the left hand side of an assignment``` or something -- try to pick off a few common cases, with a fallback case.

I'm not sure the Swift book precisely defines what lvalues are, and diagnostics should not talk about lvalues. But we need to come up with some precise terminology to describe what can appear on the left hand side of =. Basically, some combination of mutable bindings, properties, and subscripts.